plugins/lookup/kustomize.py

```diff
@@ -0,0 +1,139 @@
+#
+#  Copyright 2021 Red Hat | Ansible
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = '''
+    lookup: kustomize
+
+    short_description: Build a set of kubernetes resources using a 'kustomization.yaml' file.
+
+    author:
+      - Aubin Bikouo <@abikouo>
+    notes:
+      - If both kustomize and kubectl are part of the PATH, kustomize will be used by the plugin.
+    description:
+      - Uses the kustomize or the kubectl tool.
+      - Return the result of C(kustomize build) or C(kubectl kustomize).
+    options:
+      dir:
+        description:
+        - The directory path containing 'kustomization.yaml',
+          or a git repository URL with a path suffix specifying same with respect to the repository root.
+        - If omitted, '.' is assumed.
+        default: "."
+      binary_path:
+        description:
+        - The path of a kustomize or kubectl binary to use.
+      opt_dirs:
+        description:
+        - An optional list of directories to search for the executable in addition to PATH.
+
+    requirements:
+      - "python >= 3.6"
+'''
+
+EXAMPLES = """
+- name: Run lookup using kustomize
+  set_fact:
+    resources: "{{ lookup('kubernetes.core.kustomize', binary_path='/path/to/kustomize') }}"
+
+- name: Run lookup using kubectl kustomize
+  set_fact:
+    resources: "{{ lookup('kubernetes.core.kustomize', binary_path='/path/to/kubectl') }}"
+
+- name: Create kubernetes resources for lookup output
+  k8s:
+    definition: "{{ item }}"
+  with_items:
+    "{{ lookup('kubernetes.core.kustomize', dir='/path/to/kustomization') }}"
```

Comment:
Get rid of the loop here and just apply the lookup directly to the definition. Even with multiple resources it's still only one loop item, and even if it were multiple items it's faster to just use a single definition with multiple resources.